File: Data Structures & Algorithms/Binary Search/BinarySearch.cpp
#include <bits/stdc++.h>
using namespace std;

int BinarySearch (int arr[], int sz, int target) {
    int left = 0;
    int right = sz - 1;

    while (left <= right) {
        int mid = (left + right)/2;
        if (arr[mid] == target) {
            return mid;
        }
        else if (arr[mid] < target) {
            left = mid + 1;
        }
        else if (arr[mid] > target) {
            right = mid - 1;
        }
    }

    return -1;
}

int main() {
    int arr[] = {1,2,3,4,5,6,7,8,9};
    int target = 20;
    int sz = sizeof(arr)/sizeof(arr[0]);

    int result = BinarySearch (arr, sz, target);

    if (result == -1) {
        cout << "Target not found !" << endl;
    }
    else {
        cout << "Target found at the index : " << result << endl;
    }

    return 0;
}

File: Data Structures & Algorithms/Binary Search/BinarySearch.py
def BinarySearch(arr, target):
    left = 0
    right = len(arr) - 1

    while (left <= right):
        mid = (left + right) // 2
        if (arr[mid] == target):
            return mid
        elif (arr[mid] < target):
            left = mid + 1
        elif (arr[mid] > target):
            right = mid - 1

    return -1


arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
target = 20

result = BinarySearch(arr, target)

if result == -1:
    print("Target not found !")
else:
    print("Target at the index :" + str(result))
